ibis/backends/pyspark/compiler.py

```diff
@@ -1532,8 +1543,15 @@ def _get_interval_col(
             'Interval unit "{}" is not allowed. Allowed units are: '
             '{}'.format(dtype.unit, allowed_units)
         )
+
+    if isinstance(op.value, pd.Timedelta):
+        pandas_unit = _time_unit_mapping_pd_timedelta[dtype.unit]
+        value = getattr(op.value.components, pandas_unit)
```

Comment:
Is this safe? Can there be more than one components? e.g., hour and minute?

Reply:
I think it is probably better to do sth like this

```
td = op.value # pd.Timedelta
td_nanos = td.value
# Spark supports micro as the smallest unit, so if we have any nanos component, raise Error

td_micro = td_nanos / 1000

spark_inteval = F.expr(f'INTERVAL {td_mirco} MICROSECOND')
```

Reply:
This way you don't even need to keep the mapping

Reply:
Makes sense, definitely simpler than maintaining a mapping - and Pandas errors out immediately anyway if you pass a value to Timedelta that would cause an overflow, e.g. 

```
>>> pd.Timedelta('1000Y')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/_libs/tslibs/timedeltas.pyx", line 1217, in pandas._libs.tslibs.timedeltas.Timedelta.__new__
  File "pandas/_libs/tslibs/timedeltas.pyx", line 462, in pandas._libs.tslibs.timedeltas.parse_timedelta_string
  File "pandas/_libs/tslibs/timedeltas.pyx", line 514, in pandas._libs.tslibs.timedeltas.timedelta_from_spec
  File "pandas/_libs/tslibs/timedeltas.pyx", line 312, in pandas._libs.tslibs.timedeltas.cast_from_unit
OverflowError: Python int too large to convert to C long
```

Reply:
Great!